### PR TITLE
feat(site): show component examples after main heading

### PR DESCRIPTION
**Pull request by JoshAtITHAKA**
**This change:** (check at least one)

- [ ] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [x] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?

**What does this change address?**
Resolves: #179 

Displays working code example after main headings of pages as outlined in: https://github.com/ithaka/pharos/issues/179

**How does this change work?**
Relocation of elements on the documentation pages along with an added topMargin property

**Additional context**
Add any other context here.

### CODE REVIEW

File: .changeset/thin-clouds-crash.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@ithaka/pharos-site': patch
+---
+
+show component above doc page
```
Comment (daneah) [inline .changeset/thin-clouds-crash.md:5]:
```suggestion
Show live component examples at the top of the documentation
```

File: packages/pharos-site/static/guidelines/modal.docs.mdx [hunk 2]
```diff
@@ -10,7 +10,39 @@ import BestPractices from '../../src/components/statics/BestPractices.tsx';
...
 
-<PageSection title="Usage">
+## Examples
+
+```jsx live
+<>
+  <PharosButton
+    onClick={(e) => {
+      e.target.focus();
+      const modal = document.querySelector('pharos-modal');
```
Comment (Niznikr) [inline packages/pharos-site/static/guidelines/modal.docs.mdx:20]:
```suggestion
      const modal = document.querySelector('[data-pharos-component="PharosModal"]');
```

File: packages/pharos-site/static/guidelines/modal.docs.mdx [hunk 3]
```diff
@@ -10,7 +10,39 @@ import BestPractices from '../../src/components/statics/BestPractices.tsx';
...
+  <PharosModal header="Pharos modal" footer-divider>
+    <p>I am a modal</p>
+    <PharosButton slot="footer" type="button" variant="secondary" data-modal-close>
+      Cancel
+    </PharosButton>
+    <PharosButton
+      slot="footer"
+      type="button"
+      onClick={(e) => {
+        const modal = document.querySelector('pharos-modal');
```
Comment (Niznikr) [inline packages/pharos-site/static/guidelines/modal.docs.mdx:35]:
```suggestion
        const modal = document.querySelector('[data-pharos-component="PharosModal"]');
```

**Review by daneah — APPROVED** [review]
This looks alright to me—there are a couple things for future improvement:

* The gap between the main page heading and the examples seems rather big and is worth revisiting (don't think you introduced the issue)
* Some of the component examples are rather long, and maybe we'd do better to put the examples and the docs for the component into some kind of tabbed view (could probably readily use `pharos-tab` for this)

**Review by Niznikr — APPROVED** [review]

**Pull request merged**